sphinx/ext/script_include.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2012-8 Met Office.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+"""Sphinx directive for including the output of external commands into RST."""
+
+from shlex import split as sh_split
+from subprocess import Popen, PIPE
+
+from docutils import nodes
+from docutils.parsers.rst import Directive
+from docutils.statemachine import ViewList
+
+import sphinx
+from sphinx.util.nodes import nested_parse_with_titles
+
+
+class ScriptInclude(Directive):
+    """Insert parsed RST node from command output."""
+    option_spec = {}
+    required_arguments = 1
+    optional_arguments = 1000
+
+    def run(self):
+        command = sh_split(' '.join(self.arguments[0:]))
+        stdout = Popen(command, stdout=PIPE).communicate()[0]
```

Comment:
Worth adding `stdin=open(os.devnull)` to shut off STDIN, which may cause issues, as we have discovered recently.